components/admin_console/admin_definition.jsx

```diff
@@ -4208,3 +4209,30 @@ t('admin.field_names.postEditTimeLimit');
 t('admin.field_names.restrictCreationToDomains');
 t('admin.field_names.restrictDirectMessage');
 t('admin.field_names.teammateNameDisplay');
+
+let cache = null;
+const adminDefinitionsPlugins = {};
```

Comment:
I am wondering if it would be more consistent if it can store the custom components within the list of other custom pluggable components so we don't have to do another management of this specific ones here?  i.e https://github.com/mattermost/mattermost-webapp/blob/028b0876b80446b710caa62175ec6c0825e3e48d/plugins/registry.js#L180

Reply:
Yes, that is a good point, probably makes sense to move that to the store like the rest of the other registered things. I'll try to find time for this.

Reply:
@jespino 
Agree with @marianunez here. Doing this makes testing so much harder. I've felt the pain of testing `plugins/index.js::loadPlugin` if we keep module level variables. I would vote to rely on the existing plugin component registration system as opposed to doing this.

https://github.com/mattermost/mattermost-webapp/pull/3425/files#diff-bc70cb59796b98f505db27278998ae4bR462

Reply:
Done!